tgui/packages/tgui/interfaces/HighLuminosityEyesMenu.tsx

```diff
@@ -0,0 +1,171 @@
+import { BooleanLike } from '../../common/react';
+import { useBackend } from '../backend';
+import { Button, ColorBox, Input, LabeledList, NumberInput, Section } from '../components';
+import { Window } from '../layouts';
+
+type EyeColorData = {
+  mode: BooleanLike;
+  hasOwner: BooleanLike;
+  left: string;
+  right: string;
+};
+
+type HighLuminosityEyesMenuData = {
+  eyeColor: EyeColorData;
+  lightColor: string;
+  range: number;
+};
+
+const LightColorDisplay = (props, context) => {
+  const { act, data } = useBackend<HighLuminosityEyesMenuData>(context);
+  const { lightColor } = data;
+  const light = 0;
+  return (
+    <Section title="Color">
+      <LabeledList>
+        <LabeledList.Item label="Light Color">
+          <ColorBox color={lightColor} />{' '}
+          <Button
+            icon="palette"
+            onClick={() => act('pick_color', { to_update: light })}
+            tooltip="Brings up a color pick window to change the light color."
+          />
+          <Button
+            icon="dice"
+            onClick={() => act('random_color', { to_update: light })}
+            tooltip="Randomizes the light color."
+          />
+          <Input
+            value={lightColor}
+            width={6}
+            maxLength={7}
+            onChange={(_, value) =>
+              act('enter_color', { new_color: value, to_update: light })
+            }
+          />
+        </LabeledList.Item>
+      </LabeledList>
+    </Section>
+  );
+};
+
+const RangeDisplay = (props, context) => {
+  const { act, data } = useBackend<HighLuminosityEyesMenuData>(context);
+  const { range } = data;
+  return (
+    <Section title="Range">
+      <LabeledList>
+        <LabeledList.Item label="Light Range">
+          <Button
+            icon="minus-square-o"
+            onClick={() => act('set_range', { new_range: range - 1 })}
+            tooltip="Reduces the light range."
+          />
+          <Button
+            icon="plus-square-o"
+            onClick={() => act('set_range', { new_range: range + 1 })}
+            tooltip="Increases the light range."
+          />
+          <NumberInput
+            animated
+            width="35px"
+            step={1}
+            stepPixelSize={5}
+            value={range}
+            minValue={0}
+            maxValue={5}
+            onDrag={(e, value) =>
+              act('set_range', {
+                new_range: value,
+              })
+            }
+          />
+        </LabeledList.Item>
+      </LabeledList>
+    </Section>
+  );
+};
+
+const EyeColorDisplay = (props, context) => {
+  const { act, data } = useBackend<HighLuminosityEyesMenuData>(context);
+  const { eyeColor } = data;
+  const left = 1;
+  const right = 2;
```

Comment:
these should also be an enum, easy case here,
`enum Eye {
 Left,
 Right
}`

Reply:
It's identical

Reply:
even using a const-as-const would be okay here but the codebase has kind of settled as using enums for interfacing js-to-dm.
however, even left as a const, it's better to move things to top level scope, as the component is recreating it on each render. in this case it's barely a micro op but more of a convention. somewhat related:

https://stackoverflow.com/questions/80802/does-use-of-anonymous-functions-affect-performance/81329#81329

the general rule is moving everything to the top level that doesn't explicitly depend on or reference a state etc.